### PR TITLE
Refactor DocumentOwner

### PR DESCRIPTION
**Pull request by woutslakhorst**
Part 2 of #3208 

This PR changes the `DocumentOwner`. It's method independent with a fallback to the private key store.

### CODE REVIEW

File: vdr/did_owner.go [hunk 1]
```diff
@@ -20,17 +20,81 @@ package vdr
...
+}
+
+func (m *MultiDocumentOwner) IsOwner(ctx context.Context, d did.DID) (bool, error) {
+	for _, do := range m.DocumentOwners {
+		owned, err := do.IsOwner(ctx, d)
+		if owned || err != nil {
+			return owned, err
+		}
+	}
+	return false, nil
```
[the remark this answers is not in the document]
Comment (gerardsn) [reply vdr/did_owner.go:77]: On second thought. With the current implementations this is only possible in theory. I prefer splitting this in two separate conditions so it hurts my brain a bit less

File: vdr/didweb/manager.go [hunk 2]
```diff
@@ -263,7 +238,7 @@ func (m Manager) createService(tx *gorm.DB, subjectDID did.DID, service did.Serv
 
 	_, err = didDocumentManager.CreateOrUpdate(current.DID, current.VerificationMethods, append(current.Services, sqlService))
 
-	return &service, nil
+	return &service, err
```
Comment (gerardsn) [inline vdr/didweb/manager.go:241]: missing test?

**Review by gerardsn — APPROVED** [review]
codeclimate is complaining about `go fmt` issue.

File: vdr/didweb/manager.go (hunk 2 above)
Comment (woutslakhorst) [reply vdr/didweb/manager.go:241]: probably, but will be removed in next PR anyway

**Pull request merged**